mlrun/common/model_monitoring/helpers.py

```diff
@@ -87,3 +89,95 @@ def pad_features_hist(feature_stats: FeatureStats) -> None:
     for feature in feature_stats.values():
         if hist_key in feature:
             pad_hist(Histogram(feature[hist_key]))
+
+
+def get_model_endpoints_creation_task_status(
+    server,
+    logging_method: typing.Callable,
+) -> tuple[
+    mlrun.common.schemas.BackgroundTaskState, typing.Optional[mlrun.utils.now_date]
+]:
+    background_task = None
+    background_task_state = mlrun.common.schemas.BackgroundTaskState.running
+    background_task_check_timestamp = None
+    try:
+        background_task = mlrun.get_run_db().get_project_background_task(
+            server.project, server.model_endpoint_creation_task_name
+        )
+        background_task_check_timestamp = mlrun.utils.now_date()
+        logging_method(
+            server, background_task.status.state, background_task_check_timestamp
+        )
+        background_task_state = background_task.status.state
+    except mlrun.errors.MLRunNotFoundError:
+        logger.warning(
+            "Model endpoint creation task not found listing model endpoints",
+            project=server.project,
+            task_name=server.model_endpoint_creation_task_name,
+        )
+    if background_task is None:
+        try:
+            model_endpoints = mlrun.get_run_db().list_model_endpoints(
+                project=server.project,
+                function_name=server.function_name,
+                function_tag=server.function_tag,
+                as_dict=True,
+            )
+            if model_endpoints:
```

Comment:
we should check also that we have all the relevant endpoints no? maybe it is enough to check the expected length..